x/distribution/client/cli/suite_test.go

```diff
@@ -678,3 +678,51 @@ func (s *CLITestSuite) TestNewFundCommunityPoolCmd() {
 		})
 	}
 }
+
+func (s *CLITestSuite) TestNewWithdrawAllTokenizeShareRecordRewardCmd() {
+	val := testutil.CreateKeyringAccounts(s.T(), s.kr, 1)
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		expectedCode uint32
+		respType     proto.Message
+	}{
+		{
+			"valid transaction of withdraw tokenize share record reward",
+			[]string{
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastSync),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(10))).String()),
+			},
+			false, 0, &sdk.TxResponse{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.NewWithdrawAllTokenizeShareRecordRewardCmd()
+
+			out, err := clitestutil.ExecTestCLICmd(s.clientCtx, cmd, tc.args)
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err, out.String())
+				s.Require().NoError(s.clientCtx.Codec.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code, out.String())
+			}
+		})
+	}
+}
+
+// This test requires multiple validators, if I add this test to `IntegrationTestSuite` by increasing
+// `NumValidators` the existing tests are leading to non-determnism so created new suite for this test.
+func (s *CLITestSuite) TestNewWithdrawAllRewardsGenerateOnly() {
+	// TODO add LSM test
```

Comment:
This TODO is still pending

Reply:
Good catch. This test is now located in [x/tests/e2e/distribution/withdraw_all_suite.go](https://github.com/informalsystems/cosmos-sdk/blob/feat/lsm/v0.47.x/tests/e2e/distribution/withdraw_all_suite.go#L49C42-L49C42).